docs/kubernetes/mailu/index.rst

```diff
@@ -0,0 +1,159 @@
+Install Mailu master on kubernetes
+==================================
+
+Prequisites
+-----------
+
+Structure
+~~~~~~~~~
+
+There’s chosen to have a double NGINX stack for Mailu, this way the main
+ingress can still be used to access other websites/domains on your
+cluster. This is the current structure:
+
+-  ``NGINX Ingress controller``: Listens to the nodes ports 80 & 443. We decided to keep the Ingress Controller, because he is good in what he does.
+-  ``Cert manager``: Creates automatic Lets Encrypt certificates based on an ``Ingress``-objects domain name.
+-  ``Mailu NGINX Front daemonset``: This daemonset runs in parallel with the Nginx Ingress Controller and only listens on all E-mail specific ports (25, 110, 143, 587,...)
+-  ``Mailu components``: All Mailu components are split into separate files to make them more handy to use
+
+What you need
+~~~~~~~~~~~~~
+
+-  A working Kubernetes cluster (tested with 1.10.5)
+-  A working `cert-manager`_ installation
+-  A working nginx-ingress controller needed for the lets-encrypt
+   certificates. You can find those files in the ``nginx`` subfolder
+
+Cert manager
+^^^^^^^^^^^^
+
+The ``Cert-manager`` is quite easy to deploy using Helm when reading the
+`docs`_. After booting the ``Cert-manager`` you’ll need a
+``ClusterIssuer`` which takes care of all required certificates through
+``Ingress`` items. An example:
+
+.. code:: yaml
+
+   apiVersion: certmanager.k8s.io/v1alpha1
+   kind: ClusterIssuer
+   metadata:
+     name: letsencrypt-prod
+   spec:
+     acme:
+       email: something@example.com
+       http01: {}
+       privateKeySecretRef:
+         key: ""
+         name: letsencrypt-stage
+       server: https://acme-v02.api.letsencrypt.org/directory
```

Comment:
I think a cluster issuer isn't needed unless you want to make cert-manager provide certificates across kubernetes namespaces. An `Issuer` of some kind is enough if it is created in the same namespace as the Certificate resource as far as I know.

Reply:
But perhaps it is quite common practice to utilize one single ClusterIssuer, so not changing anything may be fine.

Reply:
This was my knowledge background, I'm not yet experienced using cert-manager:
https://www.youtube.com/watch?v=etC5d0vpLZE&t=0s&list=PLShDm2AZYnK3cWZpOjV7nOpL7plH2Ztz0&index=5

Reply:
@consideRatio You're completely right, a standard `Issuer` should do the trick as well for the `mailserver` namespace. There are two main reasons why I choose for the `clusterIssuer`
- Many people will run a website along their mailing services. Hopefully in a different namespace. Therefor they can use the same `clusterIssuer` in order to obtain their certificates.
- My setup runs many websites and it just has a single `clusterIssuer` present. Depending on the *correct* `RBAC` that is set up, there should be no problem using this.

Reply:
It depends completely on everyone's personal setup, but can be easily changed by every user. That's why I provided the YAML in the docs. But this should work in most setups, and could be easily implemented if there's already a `clusterIssuer` present

Reply:
Ah thanks I learn actively about cert-manager right.
What about the following code btw?

```yaml
       privateKeySecretRef:
         key: "" # why an empty string here?
         name: letsencrypt-stage # this is named staging, but the server below is not for staging (should be acme-staging-v02 i think)
       server: https://acme-v02.api.letsencrypt.org/directory
```

I

Reply:
- `key`-field seems to be removed now indeed. It is an older field that could reference an ACME provider which shared a private key with you. According to their [docs](http://docs.cert-manager.io/en/latest/reference/issuers.html), this field indeed doesn't exist anymore. I will remove it!
- `name`: That's indeed an error when adapting all the fields for the PR. This field just slipped my attention ;-)

Thanks for pointing this out! Commit will follow soon